lampshade.go

```diff
@@ -295,13 +295,16 @@ type StatsTracking interface {
 	EMARTT() time.Duration
 }
 
+// DialFN is a function that dials the server
+type DialFN func() (net.Conn, error)
+
 // Dialer is a dialer
 type Dialer interface {
 	StatsTracking
 
-	Dial() (net.Conn, error)
+	Dial(dial DialFN) (net.Conn, error)
```

Comment:
Passing in our dialFN on each dial makes it easier to plumb this into flashlight, in particular it allows us to preserve our op context more easily.